src/buffer/out/textBuffer.cpp

```diff
@@ -1150,9 +1150,25 @@ std::string TextBuffer::GenHTML(const TextAndColor& rows, const int fontHeightPo
                 const auto writeAccumulatedChars = [&](bool includeCurrent) {
                     if (col > startOffset)
                     {
-                        // note: this should be escaped (for '<', '>', and '&'),
-                        // however MS Word doesn't appear to support HTML entities
-                        htmlBuilder << ConvertToA(CP_UTF8, std::wstring_view(rows.text.at(row)).substr(startOffset, col - startOffset + includeCurrent));
+                        const auto unescapedText = std::wstring_view(rows.text.at(row)).substr(startOffset, col - startOffset + includeCurrent);
+                        for (const auto c : unescapedText)
+                        {
+                            switch (c)
+                            {
+                            case '<':
+                                htmlBuilder << "&lt;";
+                                break;
+                            case '>':
+                                htmlBuilder << "&gt;";
+                                break;
+                            case '&':
+                                htmlBuilder << "&amp;";
+                                break;
+                            default:
+                                htmlBuilder << ConvertToA(CP_UTF8, std::wstring_view(&c, 1));

```

Comment:
This seems like it might split up surrogate pairs. Is there a version of this algorithm that could be used on spans of text up-to-but-excluding-`<>&` instead of going wchar_t by wchar_t?

Reply:
Ah yes, you're right. Imma rework that